tests/test_widget.py

```diff
@@ -26,3 +32,29 @@ def test_widget_set_visible_invalid_string():
         widget.visible = "nope! no widget for me!"
 
     assert widget.visible
+
+
+@pytest.mark.parametrize(
+    "size_input, expectation",
+    [
+        (None, does_not_raise()),
+        (10, does_not_raise()),
+        (10.0, does_not_raise()),
+        (10.2, does_not_raise()),
+        (Scalar(100, Unit.CELLS, Unit.WIDTH), does_not_raise()),
+        (Scalar(10.2, Unit.CELLS, Unit.WIDTH), does_not_raise()),
+        ("10", does_not_raise()),
+        # And now for some common types we don't handle...
+        ("a", pytest.raises(StyleValueError)),
```

Comment:
Let's not try every possible data type here of course, but just a few ones should allow us to check that the type guard we have in `_style_properties.__set__` does it job? :slightly_smiling_face: